### PR TITLE
Bu/missing ref: Workflow for missing references

### PR DESCRIPTION
**Pull request by g-m-taylor**
#### Tests
Does this PR include tests for the changes introduced?
- [x] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: src/app/features/bulk-upload-v2/bulk-upload-references/missing-workplace-references/missing-workplace-references-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    protected alertService: AlertService,
+  ) {
+    super(errorSummaryService, formBuilder, alertService, backService, router);
+  }
+
+  ngOnInit(): void {
+    this.setBackLink(this.return);
+    this.primaryWorkplace = this.establishmentService.primaryWorkplace;
+    this.establishmentsWithMissingReferences = this.activatedRoute.snapshot.data.nextWorkplace.establishmentList;
+    this.references = filter(this.activatedRoute.snapshot.data.workplaceReferences, (reference: Workplace) => {
```
Comment (aaron-russell) [inline src/app/features/bulk-upload-v2/bulk-upload-references/missing-workplace-references/missing-workplace-references-page.component.ts:52]: If this is in 2 components, I would create a function in the parent directive, same with staff and just run the function.

File: src/app/features/bulk-upload-v2/bulk-upload-sidebar/bulk-upload-related-content/bulk-upload-related-content.component.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  templateUrl: './bulk-upload-related-content.component.html',
+})
+export class BulkUploadRelatedContentComponent {
+  @Input() showAll: boolean;
+
+  constructor(private authService: AuthService) {}
+
+  public isAdminUser(): boolean {
+    return this.authService.isAdmin;
+  }
```
Comment (aaron-russell) [inline src/app/features/bulk-upload-v2/bulk-upload-sidebar/bulk-upload-related-content/bulk-upload-related-content.component.ts:15]: You may as well just make authService public and use authService in view

**Review by aaron-russell — CHANGES_REQUESTED** [review]

File: server/models/establishment.js [hunk 3]
```diff
@@ -960,6 +960,142 @@ module.exports = function (sequelize, DataTypes) {
...
+        }
+      });
+    }
+
+  };
+  Establishment.getEstablishmentsWithMissingWorkerRef = async function(establishmentId) {
+    const isParent = this.findOne({
+      attributes: ['isParent'],
+      where: { id: establishmentId }
+    });
```
[the remark this answers is not in the document]
Comment (duncanc19) [reply server/models/establishment.js]: The request only seems to pass in the establishment ID but as this is used in three places I extracted it out into a function

[the remark this answers is not in the document]
Comment (duncanc19) [reply server/models/establishment.js]: Scrap that, I've changed it to use isParent from the request, I couldn't see it when I looked

**Review by aaron-russell — APPROVED** [review]

**Pull request merged**